src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -1832,9 +1829,10 @@ private TypeArray CreateParameterArray(MemberInfo associatedInfo, ParameterInfo[
                 types.Add(GetTypeOfParameter(p, associatedInfo));
             }
 
-            MethodInfo mi = associatedInfo as MethodInfo;
 
-            if (mi != null && (mi.CallingConvention & CallingConventions.VarArgs) == CallingConventions.VarArgs)
+            if (associatedInfo is MethodInfo mi
```

Comment:
nit:
`if (associatedInfo is MethodInfo mi && (mi.CallingConvention & CallingConventions.VarArgs) != 0 ||
     associatedInfo is ConstructorInfo ci && (ci.CallingConvention & CallingConventions.VarArgs) != 0)`

Reply:
Better yet, I had some notion in my head when I wrote that that `CallingConvention` wasn't the same method in each case, but it's really from `MethodBase` so I can just look for that.